cfg/config_test.go

```diff
@@ -1,71 +1,56 @@
 package cfg
 
 import (
-	"io/ioutil"
-	"os"
-	"path/filepath"
 	"testing"
 )
 
-func TestLoadConfig(t *testing.T) {
-	_, err := LoadConfig("foobar")
-	if err == nil {
-		t.Error("Loading an invalid config file should return an error")
-	}
-
-	conf, err := LoadConfig(filepath.Join("testdata", "beehive.conf"))
+func TestNew(t *testing.T) {
+	conf, err := New("/foobar")
 	if err != nil {
-		t.Error("Error loading config file fixture")
+		panic(err)
 	}
-
-	if conf.Bees[0].Name != "echo" {
-		t.Error("The first bee should be an exec bee named echo")
+	if _, ok := conf.Backend().(*FileBackend); !ok {
+		t.Error("Backend for '/foobar' should be a FileBackend")
 	}
-}
 
-func TestSaveConfig(t *testing.T) {
-	tmpfile, err := ioutil.TempFile("", "example")
+	conf, err = New("file:///foobar")
 	if err != nil {
-		t.Error("Could not create temp file")
+		panic(err)
+	}
+	if _, ok := conf.Backend().(*FileBackend); !ok {
+		t.Error("Backend for 'file:///foobar' should be a FileBackend")
 	}
-	defer os.Remove(tmpfile.Name()) // clean up
 
-	testConfPath := filepath.Join("testdata", "beehive.conf")
-	testConf, err := LoadConfig(testConfPath)
+	conf, err = New("mem:")
 	if err != nil {
-		t.Errorf("Failed to load config fixture %s: %v", testConfPath, err)
+		panic(err)
+	}
+	if _, ok := conf.Backend().(*MemBackend); !ok {
+		t.Error("Backend for 'mem:' should be a MemoryBackend")
 	}
+}
 
-	configFile := tmpfile.Name()
-	err = SaveConfig(configFile, testConf)
+func TestSetURL(t *testing.T) {
```

Comment:
Can't harm, but as I see it this is already covered by `TestNew`, as `New` calls `SetURL`?

Reply:
Indeed. Removed.